.github/workflows/jest-coverage-changes.yml

```diff
@@ -0,0 +1,31 @@
+name: Jest Coverage - changed files
+
+on:
+  pull_request:
+    branches: develop
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: "refs/heads/develop"
+          token: ${{ secrets.GITHUB_TOKEN }} # Provide the GitHub token for authentication
```

Comment:
for this - fetch-depth: 0 fetches everything, but we need only the target branch & current branch to be fetched and checkout. (we thought this would optimize this)

hence I have done -

target branch (develop):
```
 with:
    ref: "refs/heads/develop"
```
current branch:
```
- name: Fetch branch
    run: git fetch origin ${{ github.event.pull_request.head.ref }}
what your thought on this.

```